src/Impl/AmqpMessage.php

```diff
@@ -50,15 +50,16 @@ class AmqpMessage implements InteropAmqpMessage
      * @param string $body
      * @param array  $properties
      * @param array  $headers
+     * @param bool   $redelivered
+     * @param int    $flags
      */
-    public function __construct($body = '', array $properties = [], array $headers = [])
+    public function __construct($body = '', array $properties = [], array $headers = [], $redelivered = false, $flags = self::FLAG_NOPARAM)
```

Comment:
The constructor might become too big, It misses a routing key, maybe something else. I dont like fat constructors.